packages/appstore/src/components/modals/real-wallets-upgrade/real-wallets-upgrade.tsx

```diff
@@ -1,161 +1,133 @@
+                {localize('Next')}
+            </Button>
+        </Modal.Footer>
+    );
 
-    const StepComponent = () => {
-        return (
-            <React.Fragment>
-                {step_config.map((step, index) => {
-                    if (index === currentStep - 1) {
-                        return (
-                            <WalletSteps
-                                key={index}
-                                icon={step?.icon}
-                                title={step?.title}
-                                description={step?.description}
-                                bullets={step?.bullets || []}
-                            />
-                        );
-                    }
-                    return null;
-                })}
-            </React.Fragment>
-        );
-    };
+    const InitialFooter = (
+        <Modal.Footer className='wallet-steps__footer' has_separator>
+            <Button secondary large className='wallet-steps__footer-button' onClick={handleClose}>
+                {localize('Maybe later')}
+            </Button>
+            <Button primary large className='wallet-steps__footer-button' onClick={handleNext}>
+                {localize('Next')}
+            </Button>
+        </Modal.Footer>
+    );
 
-    const validateClickOutside = (e: MouseEvent) => {
-        // anywhere outside the modal should close the modal
-        if (wallets_upgrade_ref.current && !wallets_upgrade_ref.current.contains(e.target as Node)) {
-            return false;
-        }
-        return true;
-    };
+    const EndFooter = (
+        <Modal.Footer className='wallet-steps__footer' has_separator>
+            <Button secondary large className='wallet-steps__footer-button' onClick={handleBack}>
+                {localize('Back')}
+            </Button>
+            <Button primary large className='wallet-steps__footer-button' disabled={!is_disabled} onClick={handleClose}>
+                {localize('Upgrade to Wallets')}
+            </Button>
+        </Modal.Footer>
+    );
+
+    //  TODO: Add the wallet steps right here (͠≖ ͜ʖ͠≖)👌
+    const WalletSteps = [
+        {
+            name: 'intro_wallets',
+            component: <WalletsIntro is_eu={is_eu} current_step={0} />,
+            footer: InitialFooter,
+        },
+        {
+            name: 'intro_wallets',
+            component: <WalletsIntro is_eu={is_eu} current_step={1} />,
+        },
+        {
+            name: 'intro_wallets',
+            component: <WalletsIntro is_eu={is_eu} current_step={2} />,
+        },
+        {
+            name: 'ready_to_upgrade',
+            component: <ReadyToUpgradeWallets is_eu={is_eu} toggleCheckbox={toggleCheckbox} />,
+            footer: EndFooter,
+        },
+    ];
 
-    useOnClickOutside(wallets_upgrade_ref, handleClose, validateClickOutside);
+    const ModalContent = WalletSteps[current_step]?.component;
+    const ModalFooter = WalletSteps[current_step]?.footer || DefaultFooter;
```

Comment:
@aizad-deriv Would you wrap them with `useMemo` please 🙏🏻

Reply:
Done 😃 👍🏼